controllers/generic_controller.go

```diff
@@ -18,9 +18,10 @@ import (
 
 type (
 	Ctx[O client.Object] struct {
-		Ctx    context.Context
-		Log    logr.Logger
-		Target O
+		Ctx           context.Context
+		Log           logr.Logger
+		Target        O
+		InMaintenance bool
```

Comment:
What is the purpose of this field? It is only set, but never read.
Does it indicate to controller implementation whether the current reconciliation is within the maintenance time window?
If yes, would it make sense to rename it to `WithinMaintenance` or something similar?

Reply:
This PR is preparing this field for #43 where we want to only update a firewall-set in the Gardener maintenance time window for OS image updates. So yes, your assumption is correct. I renamed the field.